eZ/Publish/API/Repository/Tests/ContentServiceTest.php

```diff
@@ -2163,6 +2163,90 @@ public function testUpdateContentMetadataThrowsInvalidArgumentExceptionOnNoMetad
         $contentService->updateContentMetadata($contentInfo, $contentMetadataUpdateStruct);
     }
 
+    /**
+     * @covers \eZ\Publish\API\Repository\ContentService::updateContentMetadata
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function testUpdateContentAlwaysAvailable()
+    {
+        $repository = $this->getRepository();
+        $contentService = $repository->getContentService();
+
+        $folder = $this->createFolder(['eng-GB' => 'Folder'], 2);
+
+        $contentMetadataUpdate = $contentService->newContentMetadataUpdateStruct();
+        $contentMetadataUpdate->alwaysAvailable = !$folder->contentInfo->alwaysAvailable;
+        $contentService->updateContentMetadata($folder->contentInfo, $contentMetadataUpdate);
+
+        $reloadedFolder = $contentService->loadContent($folder->contentInfo->id);
+        self::assertEquals(
+            $contentMetadataUpdate->alwaysAvailable,
+            $reloadedFolder->contentInfo->alwaysAvailable
+        );
+    }
+
+    /**
+     * @covers \eZ\Publish\API\Repository\ContentService::updateContentMetadata
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    public function testUpdateContentMainTranslation()
+    {
+        $repository = $this->getRepository();
+        $contentService = $repository->getContentService();
+        $contentTypeService = $repository->getContentTypeService();
+        $locationService = $repository->getLocationService();
+
+        // create Content for a type which is not always available by default
+        $contentCreate = $contentService->newContentCreateStruct(
+            $contentTypeService->loadContentTypeByIdentifier('forum'),
+            'eng-US'
+        );
+        $contentCreate->setField('name', 'My Forum');
+        $content = $contentService->publishVersion(
+            $contentService->createContent(
+                $contentCreate,
+                [$locationService->newLocationCreateStruct(2)]
+            )->getVersionInfo()
+        );
+        // perform sanity check
+        self::assertFalse($content->contentInfo->alwaysAvailable);
+
+        $newMainTranslation = 'eng-GB';
```

Comment:
perhaps it is out of scope to extract `eng-GB` in the whole file but could you extract it for new methods?